std/algorithm/sorting.d

```diff
@@ -4462,3 +4465,260 @@ shapes. Here's a non-trivial example:
     }
     assert(n == 60);
 }
+
+/** This function transform `deci` value into a value in the factorial number
+system stored in `fac`.
+
+A factorial number is constructed as:
+fac[0] * 0! + fac[1] * 1! + ... fac[20] * 20!
+
+Params:
+    deci = The value to convert into the factorial number system.
+    fac = The array to store the factorial number. The array is of size 21 as
+        `ulong.max` requires 21 digits in the factorial number system.
+    idx = A variable storing the number of digits of the factorial number
+        stored in `fac`.
+
+See_Also:
+    $(HTTP en.wikipedia.org/wiki/Factorial_number_system,
+        Factorial Number System
+    )
+*/
+void decimalToFactorial(ulong deci, ref ubyte[21] fac, ref size_t idx)
```

Comment:
`deci` is awkward and unusual, just use `decimal`
Return the size instead of taking it by ref